orangecontrib/infrared/widgets/owfft.py

```diff
@@ -0,0 +1,594 @@
+import sys
+import numpy as np
+
+from PyQt4.QtGui import QGridLayout
+
+import Orange.data
+from Orange.widgets.widget import OWWidget
+from Orange.widgets import gui, settings
+
+def build_spec_table(wavenumbers, intensities):
+    """
+    Converts numpy arrays of wavenumber and intensity into an
+    Orange.data.Table spectra object.
+
+    Args:
+        wavenumbers (np.array): 1D array of wavenumbers
+        intensities (np.array): 2D array of (multi-spectra) intensities
+
+    Returns:
+        table: Orange.data.Table object, spectra format
+    """
+
+    # Add dimension to 1D array if necessary
+    if intensities.ndim == 1:
+        intensities = intensities[None,:]
+
+    # Convert the wavenumbers array into a list of ContinousVariables
+    wn_vars = [Orange.data.ContinuousVariable.make(repr(wavenumbers[i]))
+                for i in range(wavenumbers.shape[0])]
+
+    # Build an Orange.data.Domain object with wn_vars as
+    # independant variables (or "attributes" as Orange calls them)
+    domain = Orange.data.Domain(wn_vars)
+
+    # Finally, build the table using the damain and intensity arrays:
+    table = Orange.data.Table.from_numpy(domain, intensities)
+    return table
+
+class OWFFT(OWWidget):
+    # Widget's name as displayed in the canvas
+    name = "FFT"
+
+    # Short widget description
+    description = (
+        "Performs Fast Fourier Transform on an interferogram, including "
+        "zero filling, apodization and phase correction.")
+
+    # An icon resource file path for this widget
+    # (a path relative to the module where this widget is defined)
+    icon = "icons/fft.svg"
+
+    # Define inputs and outputs
+    inputs = [("Interferogram", Orange.data.Table, "set_data")]
+    outputs = [("Spectra", Orange.data.Table)]
+
+    # Define widget settings
+    laser_wavenumber = settings.Setting(15797.337544)
```

Comment:
We should keep the doors open for NanoIR instruments. They don't use lasers to measure the mirror position.